### PR TITLE
Optimize long sequence inference memory

### PR DESCRIPTION
**Pull request by oahzxl**
Use chunk, code optimization and heterogeneous computing to reduce memory usage of long sequence inference.

### CODE REVIEW

**Comment by oahzxl** [comment]
can infer sequence 5000 on 80G A100 now

File: fastfold/model/nn/embedders.py [hunk 1]
```diff
@@ -137,100 +138,101 @@ def forward(
...
-
-        # [*, N, N, no_bins]
-        d = ((d > squared_bins) * (d < upper)).type(x.dtype)
-
-        # [*, N, N, C_z]
-        d = self.linear(d)
-        z_update = d + self.layer_norm_z(z)
-
-        return m_update, z_update
+# class RecyclingEmbedder(nn.Module):
```
Comment (Gy-Lu) [inline fastfold/model/nn/embedders.py:141]: is this class useless?

Comment (oahzxl) [reply fastfold/model/nn/embedders.py]: Yes, I move it to ops.py. Remove it now.

**Comment by Shenggan** [comment]
LGTM, support multimer model in new PR

**Review by Shenggan — APPROVED** [review]

**Review by Gy-Lu — APPROVED** [review]

**Pull request merged**